src/compiler/emitter.ts

```diff
@@ -2709,6 +2709,22 @@ var __param = (this && this.__param) || function (paramIndex, decorator) {
                     }
                     else {
                         if (node.flags & NodeFlags.Default) {
+                            if (compilerOptions.module === ModuleKind.CommonJS || compilerOptions.module === ModuleKind.AMD || compilerOptions.module === ModuleKind.UMD) {

```

Comment:
you also want to check that this module does not have any exported name called "__esModule" so as not to step on it.